wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/ServersChecker.java

```diff
@@ -195,10 +199,21 @@ ServerChecker doCreateChecker(URL url, String serverRef) {
     // workaround needed because terminal server doesn't have endpoint to check it readiness
     if ("terminal".equals(serverRef)) {
       return new TerminalHttpConnectionServerChecker(
-          url, machineName, serverRef, 3, 180, TimeUnit.SECONDS, timer);
+          url, machineName, serverRef, 3, 180, 1, TimeUnit.SECONDS, timer);
+    }
+    if ("wsagent/http".equals(serverRef)) {
+      return new HttpConnectionServerChecker(
+          url,
+          machineName,
+          serverRef,
+          3,
+          180,
+          wsagentPingSuccessThreshold,
+          TimeUnit.SECONDS,
+          timer);
     }
     // TODO do not hardcode timeouts, use server conf instead
     return new HttpConnectionServerChecker(
-        url, machineName, serverRef, 3, 180, TimeUnit.SECONDS, timer);
+        url, machineName, serverRef, 3, 180, 1, TimeUnit.SECONDS, timer);
```

Comment:
In case wsagent server is not first to boot server in a machine we can face a situation when success threshold for other agents needs to be bigger than 1 too. So, I suggest using the same threshold from the property for all the servers for now. And if so property `che.workspace.agent.dev.ping_success_threshold` can be more wsagent and dev machine agnostic, e.g. `che.workspace.server.ping_success_threshold`. WDYT?

Reply:
@garagatyi sounds relevant. Let me update the PR